chore: move files from docker-base-images into separate repository

abtris opened the pull request:
- Try automate releasing docker images with dredd release
- I added cron to run master every week, docker image will be always in max 1 week old.

------

Closes https://github.com/apiaryio/dredd/issues/923, closes https://github.com/apiaryio/dredd/issues/1106

Commit:
chore: move files from docker-base-images into separate repository

kylef commented on package.json:
```diff
@@ -0,0 +1,19 @@
+    "url": "git+https://github.com/apiaryio/dredd-docker.git"
+  },
+  "author": "",
+  "license": "MIT",
```
MIT license, but no license file in this changeset.

abtris replied:
already in master

nadade commented on Dockerfile:
```diff
@@ -0,0 +1,16 @@
+# https://hub.docker.com/r/mhart/alpine-node/
```
not using mhart/alpine-node, can you remove the comment?

nadade commented on Dockerfile:
```diff
@@ -0,0 +1,16 @@
+# If you have native dependencies, you'll need extra tools
+
+MAINTAINER  Apiary <sre@apiary.io>
+ENV REFRESHED_AT 2018-04-17
```
and update the date or remove the env var?

honzajavorek committed:
chore: use double quotes consistently

honzajavorek committed:
chore: use extensions consistently

honzajavorek committed:
chore: forget about latest vs. stable

Dredd does not use the tags anymore and it's there only for historical
purposes. Let's not complicate our lives, $(npm view dredd version) is
just fine.

honzajavorek committed:
chore: address @nadade's code review comments

honzajavorek committed:
chore: remove the package.json as it's redundant now

honzajavorek committed:
docs: revamp the README

Not documenting the workflow here as I want to have it in the Dredd docs.

honzajavorek committed:
chore: try to strip down the config

honzajavorek committed:
chore: make sure the 'stable' tag gets built as well

Sure, we don't use it, but it's our legacy and it could be confusing
for someone if this tag is lagging behind all the other tags and versions.

honzajavorek commented:
@abtris @kylef @nadade Would you please re-review this? I made a lot of changes, so consider this to be my PR now 😄 I'm sorry @abtris for hijacking your PR.

honzajavorek committed:
chore: fix line wrapping

abtris commented on docker_push.sh:
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
-DREDD_VERSION=`./scripts/getLatestStableNodePackage dredd`
+DREDD_VERSION=$(npm view dredd version)
```
nice!

honzajavorek replied:
@abtris If you wanted to use the `dist-tags`, it looks like this for `stable`:

```shell
$ npm view dredd dist-tags --json | grep 'stable' | python -c 'import sys, re; print(re.search(r": \"([^\"]+)", sys.stdin.read()).group(1))'
5.2.0
```

Or like this for `latest`:

```shell
$ npm view dredd dist-tags --json | grep 'latest' | python -c 'import sys, re; print(re.search(r": \"([^\"]+)", sys.stdin.read()).group(1))'
5.2.0
```

But that's not necessary for Dredd at the moment as we're not doing anything with the tags. Dredd currently always automatically releases the latest version to both `latest` and `stable`.

abtris replied:
👍

abtris replied:
or, without python.

```
npm view dredd dist-tags --json | grep 'stable' | cut -d ":" -f2 | tr -d '"'
```

honzajavorek replied:
Nice. I'm not that skilled in bash and when I run out of ideas, I use Python 😛

abtris commented:
@honzajavorek I can't approve my PR

honzajavorek commented:
@abtris You can write "LGTM" here 😏

abtris commented:
ok, LGTM

kylef commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+  - "bash build.sh"
+deploy:
+  provider: "script"
+  script: "bash docker_push.sh"
```
Seems like there is room for a nasty timing issue, since both build.sh and docker_push.sh use `DREDD_VERSION=$(npm view dredd version)` to get the Dredd version, what if the latest version were to change between the execution of the two scripts? I think it would be better to pass it to both scripts and compute once.

honzajavorek replied:
https://github.com/apiaryio/dredd-docker/pull/2

kylef commented on Dockerfile:
```diff
@@ -0,0 +1,13 @@
+RUN apk add --no-cache make gcc g++ python git
+RUN npm config set loglevel error
+RUN npm install dredd@${DREDD_VERSION}
+RUN apk del make gcc g++ python
```
One tip to make the docker image much smaller would be to mimimise the amount of layers it has. Each run invocation creates a seperate layer in the image, so even though you are deleting the packages they are still in previous two layer. You can make all the commands happen in a single layer to prevent this. For example:

```
RUN apk add --no-cache make gcc g++ python git && \
  npm config set loglevel error && \
  npm install dredd@${DREDD_VERSION} && \
  apk del make gcc g++ python
```

honzajavorek replied:
https://github.com/apiaryio/dredd-docker/pull/2
